packages/hash/frontend/src/components/hooks/useCreateSubPage.ts

```diff
@@ -0,0 +1,62 @@
+import { useMutation } from "@apollo/client";
+import { useRouter } from "next/router";
+import { useCallback } from "react";
+import {
+  CreatePageMutation,
+  CreatePageMutationVariables,
+  SetParentPageMutation,
+  SetParentPageMutationVariables,
+} from "../../graphql/apiTypes.gen";
+import { getAccountPagesTree } from "../../graphql/queries/account.queries";
+import { createPage, setParentPage } from "../../graphql/queries/page.queries";
+
+export const useCreateSubPage = (accountId: string) => {
+  const router = useRouter();
+
+  const [createPageFn, { loading: createPageLoading }] = useMutation<
+    CreatePageMutation,
+    CreatePageMutationVariables
+  >(createPage);
+
+  const [setParentPageFn, { loading: setParentPageLoading }] = useMutation<
+    SetParentPageMutation,
+    SetParentPageMutationVariables
+  >(setParentPage, {
+    awaitRefetchQueries: true,
+    refetchQueries: ({ data }) => [
+      {
+        query: getAccountPagesTree,
+        variables: { accountId: data!.setParentPage.accountId },
```

Comment:
Fetch results `data` is nullable for some reason. I didn't really write this code, I just separated the createSubpage from the createPage hook, for keeping it simpler to track loading states. Would you rather have `?` instead of the `!` ?